contracts/cw20-staked-balance-voting/src/contract.rs

```diff
@@ -0,0 +1,257 @@
+            if initial_supply.is_zero() {
+                return Err(ContractError::InitialBalancesError {});
+            }
+
+            STAKING_CONTRACT_CODE_ID.save(deps.storage, &staking_code_id)?;
+            STAKING_CONTRACT_UNSTAKING_DURATION.save(deps.storage, &unstaking_duration)?;
+
+            let msg = WasmMsg::Instantiate {
+                admin: Some(info.sender.to_string()),
+                code_id,
+                msg: to_binary(&cw20_base::msg::InstantiateMsg {
+                    name,
+                    symbol,
+                    decimals,
+                    initial_balances,
+                    mint: Some(cw20::MinterResponse {
+                        minter: info.sender.to_string(),
+                        cap: None,
+                    }),
+                    marketing,
+                })?,
+                funds: vec![],
+                label,
+            };
+            let msg = SubMsg::reply_on_success(msg, INSTANTIATE_TOKEN_REPLY_ID);
+
+            Ok(Response::default()
+                .add_attribute("action", "instantiate")
+                .add_attribute("token", "new_token")
+                .add_submessage(msg))
+        }
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    _deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {}
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::TokenContract {} => query_token_contract(deps),
+        QueryMsg::StakingContract {} => query_staking_contract(deps),
+        QueryMsg::VotingPowerAtHeight { address, height: _ } => {
+            query_voting_power_at_height(deps, env, address)
+        }
+        QueryMsg::TotalPowerAtHeight { height: _ } => query_total_power_at_height(deps, env),
+        QueryMsg::Info {} => query_info(deps),
+    }
+}
+
+pub fn query_token_contract(deps: Deps) -> StdResult<Binary> {
+    let token = TOKEN.load(deps.storage)?;
+    to_binary(&token)
+}
+
+pub fn query_staking_contract(deps: Deps) -> StdResult<Binary> {
+    let staking_contract = STAKING_CONTRACT.load(deps.storage)?;
+    to_binary(&staking_contract)
+}
+
+pub fn query_voting_power_at_height(deps: Deps, env: Env, address: String) -> StdResult<Binary> {
+    let staking_contract = STAKING_CONTRACT.load(deps.storage)?;
+    let address = deps.api.addr_validate(&address)?;
+    let res: stake_cw20::msg::StakedBalanceAtHeightResponse = deps.querier.query_wasm_smart(
+        staking_contract,
+        &stake_cw20::msg::QueryMsg::StakedBalanceAtHeight {
+            address: address.to_string(),
+            height: Some(env.block.height),
+        },
+    )?;
+    to_binary(
+        &cw_governance_interface::voting::VotingPowerAtHeightResponse {
+            power: res.balance,
+            height: env.block.height,
```

Comment:
We likely want to use the height returned by the staked balance query here. I think that in the event that someone has staked on the same block that we query on the `height` field will be the previous block because the state change hasn't been committed yet.